bin/sendsketch_to_prokka.py

```diff
@@ -1,35 +1,90 @@
 #!/usr/bin/env python3
-import sys
+from sys import argv, exit
+import argparse
 
-sketch_file = sys.argv[1]
-stain_file = sys.argv[2]
-profile_file = sys.argv[3]
+"""This script identifies the top ranked genus in output from sendsketch.sh from BBMap.
 
-output_stain = "Not_in_list"
-output_species = "taxa"
-genus = "Multiple"
+Script was developed for internal use in the Nextflow pipeline BACTpipe.
+"""
 
-with open(sketch_file, "r") as sketch:
-    sketch_lines = sketch.readlines()
+def parse_args():
+    """Parse command line arguments.
+    """
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("-s", "--sketch",
+        required=True,
+        help="Path to sendsketch.sh output file (txt)")
+    parser.add_argument("-S", "--stain", 
+        required=True,
+        help="Path to text file containing gram staining classifications in "
+            "two-column tab separated format (Genus<TAB>Stain)")
+    parser.add_argument("-p", "--profile", 
+        required=True,
+        help="Path to TSV file with profile information")
 
-    genus_one_line = sketch_lines[3]
-    genus_one = genus_one_line.split("\t")[11].split(" ")[0]
-    species_one = genus_one_line.split("\t")[11].split(" ")[1]
+    if len(argv) < 2:
+        parser.print_help()
+        exit(1)
 
-    genus_two_line = sketch_lines[4]
-    genus_two = genus_two_line.split("\t")[11].split(" ")[0]
+    args = parser.parse_args()
 
-    if len(genus_two) == 0 or genus_one == genus_two:
-        genus = genus_one
-        output_species = species_one.rstrip()
-        with open(stain_file, "r") as stain:
-            for line in stain:
-                if line.split("\t")[0] == genus:
-                    output_stain = line.rstrip().split("\t")[1]
-    else:
-        output_stain = "Contaminated"
-print(output_stain+"\t"+genus+"\t"+output_species)
+    return args
 
-with open(profile_file, "w") as f:
-    f.writelines(output_stain + "\t" + genus + "\t" + output_species)
 
+def parse_stain_database(stain_file):
+    """Parse gram stain database into dictionary for quick lookups.
+    """
+    stain_db = {}
+    with open(stain_file, "r") as stain:
+        for line in stain:
+            genus, curr_stain = line.split("\t")
+            stain_db[genus] = curr_stain.rstrip()
+    return stain_db
+
+
+def parse_sendsketch(sketch_file, stain_db):
+    """Parse sendsketch.sh output and output if sample is contaminated or not"""
+
+    with open(sketch_file, "r") as sketch:
+        sketch_lines = sketch.readlines()
+
+        if not sketch_lines[2].startswith("WKID\t"):
+            print("ERROR: sketch file '{}' does not appear to contain valid sendsketch.sh output".format(args.sketch))
+            exit(1)
+
+        genus_one_line = sketch_lines[3]
+        try:
+            genus_one = genus_one_line.split("\t")[11].split(" ")[0]
+            species_one = genus_one_line.split("\t")[11].split(" ")[1]
+        except ValueError as e:
```

Comment:
This should except IndexError, not ValueError, my bad!

```suggestion
        except IndexError as e:
```